File: Python/Harddrive.py
# harddrive.py
from Komponen import Komponen

class Harddrive(Komponen):
    """Hard Drive class"""
    
    def __init__(self, kapasitas_gb=0, tipe_drive="", merk="", nama="", harga=0, 
                 read_speed=500, write_speed=450, tipe_komponen="Storage"):
        super().__init__(merk, nama, harga, tipe_komponen)
        self.kapasitas_gb = kapasitas_gb
        self.tipe_drive = tipe_drive
        self.read_speed = read_speed
        self.write_speed = write_speed
    
    def get_kapasitas_gb(self):
        return self.kapasitas_gb
    
    def set_kapasitas_gb(self, kapasitas_gb):
        self.kapasitas_gb = kapasitas_gb
    
    def get_tipe_drive(self):
        return self.tipe_drive
    
    def set_tipe_drive(self, tipe_drive):
        self.tipe_drive = tipe_drive
    
    def get_read_speed(self):
        return self.read_speed
    
    def set_read_speed(self, read_speed):
        self.read_speed = read_speed
    
    def get_write_speed(self):
        return self.write_speed
    
    def set_write_speed(self, write_speed):
        self.write_speed = write_speed
    
    def tampilkan_info(self):
        print(f"Storage: {self.merk} {self.nama} ({self.kapasitas_gb} GB, {self.tipe_drive})")
        print(f"    Read: {self.read_speed} MB/s, Write: {self.write_speed} MB/s, Harga: Rp{self.harga}")

File: Java/Harddrive.java
// Harddrive.java
public class Harddrive extends Komponen {
    private int kapasitasGB;
    private String tipeDrive;
    private int readSpeed;
    private int writeSpeed;
    
    public Harddrive() {
        super();
        this.kapasitasGB = 0;
        this.tipeDrive = "";
        this.readSpeed = 0;
        this.writeSpeed = 0;
    }
    
    public Harddrive(int kapasitasGB, String tipeDrive, String merk, String nama, double harga,
                     int readSpeed, int writeSpeed) {
        super(merk, nama, harga, "Storage");
        this.kapasitasGB = kapasitasGB;
        this.tipeDrive = tipeDrive;
        this.readSpeed = readSpeed;
        this.writeSpeed = writeSpeed;
    }
    
    public int getKapasitasGB() {
        return kapasitasGB;
    }
    
    public void setKapasitasGB(int kapasitasGB) {
        this.kapasitasGB = kapasitasGB;
    }
    
    public String getTipeDrive() {
        return tipeDrive;
    }
    
    public void setTipeDrive(String tipeDrive) {
        this.tipeDrive = tipeDrive;
    }
    
    public int getReadSpeed() {
        return readSpeed;
    }
    
    public void setReadSpeed(int readSpeed) {
        this.readSpeed = readSpeed;
    }
    
    public int getWriteSpeed() {
        return writeSpeed;
    }
    
    public void setWriteSpeed(int writeSpeed) {
        this.writeSpeed = writeSpeed;
    }
    
    @Override
    public void tampilkanInfo() {
        System.out.println("Storage: " + merk + " " + nama + 
                           " (" + kapasitasGB + " GB, " + tipeDrive + ")");
        System.out.println("    Read: " + readSpeed + " MB/s, Write: " + writeSpeed + 
                           " MB/s, Harga: Rp" + harga);
    }
}